kolibri/core/assets/src/views/__tests__/TimeDuration.spec.js

```diff
@@ -0,0 +1,57 @@
+import { render, screen } from '@testing-library/vue';
+import VueRouter from 'vue-router';
+import TimeDuration from '../TimeDuration.vue';
+
+const renderComponent = (props = {}) => {
+  return render(TimeDuration, {
+    props,
+    routes: new VueRouter(),
+  });
+};
+
+const MINUTE = 60;
+const HOUR = 60 * MINUTE;
+const DAY = 24 * HOUR;
+
+const testCases = [
+  // Under 2 minutes, should show seconds
+  { seconds: 0, expected: '0 seconds' },
+  { seconds: 1, expected: '1 second' },
+  { seconds: 59, expected: '59 seconds' },
+  { seconds: MINUTE, expected: '60 seconds' },
+  { seconds: 2 * MINUTE - 1, expected: '119 seconds' },
+
+  // Under 1 hour, should show minutes (rounded down)
+  { seconds: 2 * MINUTE, expected: '2 minutes' },
+  { seconds: 30 * MINUTE, expected: '30 minutes' },
+  { seconds: 30 * MINUTE + 1, expected: '30 minutes' },
+  { seconds: 30 * MINUTE + 59, expected: '30 minutes' },
+  { seconds: 59 * MINUTE, expected: '59 minutes' },
+
+  // Under 1 day, should show hours (rounded down)
+  { seconds: HOUR, expected: '1 hour' },
+  { seconds: 2 * HOUR, expected: '2 hours' },
+  { seconds: 23 * HOUR, expected: '23 hours' },
+  { seconds: 23 * HOUR + 59 * MINUTE, expected: '23 hours' },
+  { seconds: 23 * HOUR + 59 * MINUTE + 59, expected: '23 hours' },
+
+  // Over 1 day, should show days (rounded down)
+  { seconds: DAY, expected: '1 day' },
+  { seconds: 2 * DAY, expected: '2 days' },
+  { seconds: 6 * DAY, expected: '6 days' },
+  { seconds: 6 * DAY + 23 * HOUR + 59 * MINUTE + 59, expected: '6 days' },
+];
+
+describe('TimeDuration', () => {
+  testCases.forEach(({ seconds, expected }) => {
+    it(`should render ${expected} for ${seconds} seconds`, () => {
+      renderComponent({ seconds });
+      expect(screen.getByText(expected)).toBeInTheDocument();
+    });
+  });
```

Comment:
The tests here in TimeDuration are excellent! I love the approach of setting up the array of objects and iterating over them here.

This is a non-blocking note as your tests here are clear, but I wanted to share this jest global [test.each](https://jestjs.io/docs/api#1-testeachtablename-fn-timeout)

The syntax is a little bit strange, but it's a pretty neat way to make these kind of iterative tests

```javascript
// from the docs I linked
// in our case we'd use `it.each`
test.each([
  {a: 1, b: 1, expected: 2},
  {a: 1, b: 2, expected: 3},
  {a: 2, b: 1, expected: 3},
])('.add($a, $b)', ({a, b, expected}) => {
  expect(a + b).toBe(expected);
});
```

Reply:
Thanks! Will update with this utility.